app/models/course/announcement.rb

```diff
@@ -18,4 +19,18 @@ def set_defaults
   def valid_in_future?
     valid_from > DateTime.now
   end
+
+  # Get the number of unread announcements for current user in current course
+  #
+  # @param current_user [User] The login user.
+  # @param course [Course] The Course which user is visiting
+  # @return count [Integer] The number of unread announcements.
+  def self.get_unread_number(current_user, course)
+    count = -1
+    if (!current_user.nil? && !course.nil?)
```

Comment:
<code>return unless current_user && course </code>

is it better ? Then no need to declare <code>count</code>